web-ui/src/main/resources/catalog/components/edit/bounding/BoundingDirective.js

```diff
@@ -164,17 +165,16 @@
             // parse initial input coordinates to display shape
             ctrl.initValue = function() {
               if (ctrl.polygonXml) {
+                var srsName = ctrl.polygonXml.match(new RegExp("srsName=\"(EPSG:[0-9]*)\""));
+                ctrl.dataProjection = srsName && srsName.length === 2 ?
+                  srsName[1] : 'EPSG:4326';
+
                 // parse first feature from source XML & set geometry name
-                var correctedXml = ctrl.polygonXml
-                    .replace(/<gml:LinearRingTypeCHOICE_ELEMENT0>/g, '')
-                    .replace(/<\/gml:LinearRingTypeCHOICE_ELEMENT0>/g, '')
-                    .replace(/<gml:LineStringTypeCHOICE_ELEMENT1>/g, '')
-                    .replace(/<\/gml:LineStringTypeCHOICE_ELEMENT1>/g, '');
                 var geometry = gnGeometryService.parseGeometryInput(
                     ctrl.map,
-                    correctedXml,
+                    ctrl.polygonXml,
                     {
-                      crs: 'EPSG:4326',
+                      crs: ctrl.dataProjection,
```

Comment:
I'm not sure you're having the cleanest approach here. Originally the directive would always output GML in 4326 projection, a logical evolution would be to output GML in the projection selected by the user (`ctrl.currentProjection`) in the tool. Along this, it would make sense that when GML is parsed, the selected projection changes to the one defined in the GML (if any).
As such, I think we have to make the GeometryService evolve to handle CRS detection (instead of doing it in the directive) in GML.

Reply:
if you import a record with a projection not available in the GUI, then you'll alter the record limiting to the UI list of projection unless we add the record projection to the list.

Reply:
Ok, I hadn't thought of that. Maybe adding the GML projection to the list in the UI? Also I'm not sure that OpenLayers will be able to handle a projection that have not been added to the original list, as some projections must first be included with proj4js.

Another option is to put the tool in readonly mode if the GML projection is not part of the handled ones. This way the extent is still usable but not editable.